src/Microsoft.AspNet.Razor/CodeGenerators/CSharpTagHelperCodeRenderer.cs

```diff
@@ -251,69 +226,134 @@ protected virtual string GenerateUniqueId()
-                        attributeDescriptor,
-                        bufferableAttribute,
-                        attributeValueChunk,
-                        valueAccessor);
-
-                    // Execution contexts are a runtime feature.
-                    if (_designTimeMode)
+                    if (!renderedBoundAttributeNames.Add(attribute.Key))
                     {
+                        // We already rendered an attribute with the same name that was bound. First wins, skip.
                         continue;
                     }
 
-                    // We need to inform the context of the attribute value.
-                    _writer
-                        .WriteStartInstanceMethodInvocation(
-                            ExecutionContextVariableName,
-                            _tagHelperContext.ExecutionContextAddTagHelperAttributeMethodName)
-                        .WriteStringLiteral(attributeName)
-                        .WriteParameterSeparator()
-                        .Write(valueAccessor)
-                        .WriteEndMethodInvocation();
+                    // We capture the value location of each bound attribute to ensure when multiple TagHelpers request
+                    // the same attribute we don't execute the attributes value more than once.
+                    string valueLocation = null;
+
+                    foreach (var associatedDescriptor in associatedDescriptors)
+                    {
+                        // There should only be a single TagHelperAttributeDescriptor associated with the attribute.
+                        // TagHelpers cannot have multiple attribute descriptors pointing to the same property unless
+                        // they're indexed.
+                        var associatedAttributeDescriptor = associatedDescriptor.Attributes.First(
+                            attributeDescriptor => attributeDescriptor.IsNameMatch(attribute.Key));
+                        var tagHelperVariableName = GetVariableName(associatedDescriptor);
+
+                        valueLocation = RenderBoundAttribute(
+                            tagHelperVariableName,
+                            valueLocation,
+                            attribute,
+                            associatedAttributeDescriptor);
+                    }
                 }
                 else
                 {
-                    // The attribute value has already been recorded, lets retrieve it from the stored value
-                    // accessors.
-                    _writer
-                        .WriteStartAssignment(valueAccessor)
-                        .Write(htmlAttributeValues[attributeName])
-                        .WriteLine(";");
+                    RenderUnboundAttribute(attribute);
+                }
+            }
+        }
+
+        private string RenderBoundAttribute(
+            string tagHelperVariableName,
+            string valueLocation,
```

Comment:
Can't, the `valueAccessor` is used along side of the `valueLocation` https://github.com/aspnet/Razor/pull/415/files#diff-6f3af99214f3f3e7141c9d7bd2aea6c0R351